app/scripts/controllers/alert-controller.test.ts

```diff
@@ -28,231 +29,288 @@ const EMPTY_ACCOUNT = {
+            messenger.call('AlertController:getState').web3ShimUsageOrigins,
+          ).toStrictEqual({
+            testWeb3ShimUsageOrigin: 2,
+          });
+        },
+      );
     });
   });
 
   describe('selectedAccount change', () => {
-    it('should set unconnectedAccountAlertShownOrigins to {}', () => {
-      controllerMessenger.publish('AccountsController:selectedAccountChange', {
-        id: '',
-        address: '0x1234567',
-        options: {},
-        methods: [],
-        type: 'eip155:eoa',
-        metadata: {
-          name: '',
-          keyring: {
-            type: '',
-          },
-          importTime: 0,
+    it('should set unconnectedAccountAlertShownOrigins to {}', async () => {
+      await withController(
+        { state: initialState },
+        ({ controller, messenger }) => {
+          messenger.publish('AccountsController:selectedAccountChange', {
+            id: '',
+            address: '0x1234567',
+            options: {},
+            methods: [],
+            type: 'eip155:eoa',
+            metadata: {
+              name: '',
+              keyring: {
+                type: '',
+              },
+              importTime: 0,
+            },
+          });
+          expect(
+            controller.state.unconnectedAccountAlertShownOrigins,
+          ).toStrictEqual({});
+          expect(
+            messenger.call('AlertController:getState')
+              .unconnectedAccountAlertShownOrigins,
+          ).toStrictEqual({});
         },
-      });
-      expect(
-        alertController.store.getState().unconnectedAccountAlertShownOrigins,
-      ).toStrictEqual({});
-      expect(
-        controllerMessenger.call('AlertController:getState')
-          .unconnectedAccountAlertShownOrigins,
-      ).toStrictEqual({});
+      );
     });
   });
 
   describe('AlertController:getState', () => {
-    it('should return the current state of the property', () => {
-      const defaultWeb3ShimUsageOrigins = {
-        testWeb3ShimUsageOrigin: 0,
-      };
-      expect(
-        alertController.store.getState().web3ShimUsageOrigins,
-      ).toStrictEqual(defaultWeb3ShimUsageOrigins);
-      expect(
-        controllerMessenger.call('AlertController:getState')
-          .web3ShimUsageOrigins,
-      ).toStrictEqual(defaultWeb3ShimUsageOrigins);
+    it('should return the current state of the property', async () => {
+      await withController(
+        { state: initialState },
+        ({ controller, messenger }) => {
+          const defaultWeb3ShimUsageOrigins = {
+            testWeb3ShimUsageOrigin: 0,
+          };
+          expect(controller.state.web3ShimUsageOrigins).toStrictEqual(
+            defaultWeb3ShimUsageOrigins,
+          );
+          expect(
+            messenger.call('AlertController:getState').web3ShimUsageOrigins,
+          ).toStrictEqual(defaultWeb3ShimUsageOrigins);
+        },
+      );
     });
   });
 
   describe('AlertController:stateChange', () => {
```

Comment:
Hmm, is this test needed at all? We don't often test that a controller's `stateChange` event gets published when its state changes, because that functionality is provided by BaseController, and we already test it there. Perhaps we can remove it entirely?